class.jetpack.php

```diff
@@ -2804,8 +2804,10 @@ public static function activate_default_modules(
 		do_action( 'jetpack_before_activate_default_modules', $min_version, $max_version, $other_modules );
 
 		// Check each module for fatal errors, a la wp-admin/plugins.php::activate before activating
-		Jetpack::restate();
-		Jetpack::catch_errors( true );
+		if ( $send_state_messages ) {
```

Comment:
These state messages were causing issues in the unit tests where we were trying to set cookies after content had been echoed. So, I made some changes so that we're respecting `$send_state_messages`.